index.js

```diff
@@ -1,96 +1,33 @@
-const assert = require('assert')
+var pathname = require('pathname-match')
+var urlencode = require('urlencode')
+var wayfarer = require('wayfarer')
+var assert = require('assert')
 
-module.exports = serverRouter
+module.exports = ServerRouter
 
-// Server router
-// str -> fn -> any
-function serverRouter (opts, routes) {
-  if (!routes) {
-    routes = opts
-    opts = {}
-  }
+function ServerRouter (opts) {
+  if (!(this instanceof ServerRouter)) return new ServerRouter(opts)
 
-  const dft = opts.default || '/404'
-  const shouldThunk = (opts.thunk || opts.thunk === undefined) || false
-
-  assert.equal(typeof opts, 'object', 'server-router: opts should be a object')
-  assert.equal(typeof dft, 'string', 'server-router: opts.default should be a string')
-  assert.ok(Array.isArray(routes), 'server-router: routes should be an array')
-
-  const router = wayfarer(dft + '/GET')
-  match._router = router
-
-  // register tree in router
-  // tree[0] is a string, thus a route
-  // tree[0] is an array, thus not a route
-  // tree[1] is a function or object
-    // tree[2] is an array
-    // tree[2] is not an array
-  // tree[1] is an array
-  ;(function walk (tree, fullRoute) {
-    if (typeof tree[0] === 'string') {
-      var route = tree[0].replace(/^\//, '')
-    } else {
-      var rootArr = tree[0]
-    }
-
-    const children = (Array.isArray(tree[1]))
-      ? tree[1]
-      : Array.isArray(tree[2]) ? tree[2] : null
-
-    if (rootArr) {
-      tree.forEach(function (node) {
-        walk(node, fullRoute)
-      })
-    }
-
-    const cb = (typeof tree[1] === 'function')
-      ? tree[1]
-      : (typeof tree[1] === 'object')
-        ? tree[1]
-        : null
-
-    if (cb && typeof cb === 'function') {
-      const computedRoute = route
-        ? fullRoute.concat(route).join('/')
-        : fullRoute.length ? fullRoute.join('/') : route
-
-      const handler = (shouldThunk) ? thunkify(cb) : cb
-      router.on(`${computedRoute}/GET`, handler)
-    } else if (cb && typeof cb === 'object' && !Array.isArray(cb)) {
-      Object.keys(cb).forEach(function (method) {
-        assert.ok(methods.indexOf(method.toUpperCase()), `server-router: ${method} is not a valid HTTP method`)
-
-        const computedRoute = route
-          ? fullRoute.concat(route).join('/')
-          : fullRoute.length ? fullRoute.join('/') : route
-
-        const handler = (shouldThunk) ? thunkify(cb[method]) : cb[method]
-        router.on(`${computedRoute}/${method.toUpperCase()}`, handler)
-      })
-    }
-
-    if (Array.isArray(children)) {
-      walk(children, fullRoute.concat(route))
-    }
-  })(routes, [])
-
-  return match
+  opts = opts || {}
+  assert.equal(typeof opts, 'object', 'server-router: opts should be type object')
+  this._default = opts.default || '/404'
+  assert.equal(typeof this._default, 'string', 'server-router: this._default should be type string')
+  this._router = wayfarer('/GET/' + this._default.replace(/^[#/]/, ''))
```

Comment:
for readability's sake maybe define a var first and then pass to wayfarer?

while we are on this line, also, you're keeping `GET` as the default method rather than `*`?